platforms/chibios/drivers/ps2/ps2_io.c

```diff
@@ -14,6 +14,7 @@
 #    error "PS/2 data setting is required in config.h"
 #endif
 
+#ifdef PAL_MODE_OUTPUT_OPENDRAIN
```

Comment:
Despite the assertion that this would work for other platforms that don't implement open-drain GPIOs, I don't feel comfortable implementing the functionality this way.

We have a `vendor` subdirectory which already has specialisations for RP2040 -- I'd suggest dropping a copy there instead of the preprocessor conditionals here.